runner/src/internal.rs

```diff
@@ -192,8 +263,10 @@ where
 /// Run the provided step, printing out information about the execution, and returning a set of
 /// status results from the single or multiple steps that were executed.
 #[async_recursion]
-pub async fn run_step(context: &Context, step: Step) -> StepResult {
+pub async fn run_step(context: &Context, step: Step, mut run_status: Status) -> StepResult {
```

Comment:
It seems that this `run_status` is used as an output parameter, which is usually not idiomatic. As I mentioned elsewhere, there is overlap between this new Status and the StepResult value, I think it makes sense to unify them, and make it look like an actual return value.

Reply:
It is not just an output. It has to be an input as well, because it is used for annotating all the log lines with the global state of the execution. If it is changed to a return value, then it can only annotate each log line with the local execution state, which is not very interesting. Actually, in that case, it would not add any information to the log line beyond what is already there. 

So, as I stated in another comment, this variable is keeping a global state, and could be replaced with a global mutable variable. Or it could be part of the context, but I did not want to make `context` mutable. 

Let me know what you think :)